### PR TITLE
V2 revised

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Get a scoped down version of v2 revised without the transport refactor that is compatible with graphsync master

# Implementation

Essentially, this PR starts from the point in v2 where I began doing transport refactors.

The transport refactor was huge, and speculative from the beginning because we never got a second transport. I did a lot of changes, but never had confidence in the ultimate result, and more importantly, I still haven't succcessfully run the go-fil-markets integration tests with that branch. This is despite multiple attempts at it.

This PR drops the transport refactor part of data transfer v2, but ports over compatibility with graphsync and various quality of life improvements to the data transfer codebase. I have confirmed that it passes integration tests in go-fil-markets.

The path from here to shipping go-data-transfer v2 is:
1. update go-graphsync to go-libp2p v0.22, tag 
2. update this branch to go-libp2p v.22, merge, tag
3. update the go-fil-markets integration branch to go-fil-markets master and merge and tag

I think I can get these remaining tasks done quickly if this gets the ok.

### CODE REVIEW

File: channels/internal/migrations/migrations.go [hunk 1]
```diff
@@ -1,13 +1,114 @@
...
+	// Number of blocks that have been sent, including blocks that are
+	// present in more than one place in the DAG
+	SentBlocksTotal int64
+	// DataLimit is the maximum data that can be transferred on this channel before
+	// revalidation. 0 indicates no limit.
+	DataLimit uint64
+	// RequiresFinalization indicates at the end of the transfer, the channel should
+	// be left open for a final settlement
+	RequiresFinalization bool
+	// Stages traces the execution fo a data transfer.
```
Comment (willscott) [inline channels/internal/migrations/migrations.go:63]: 'of'

File: impl/events.go [hunk 2]
```diff
@@ -212,7 +211,18 @@ func (m *manager) OnResponseReceived(chid datatransfer.ChannelID, response datat
 	if response.IsPaused() {
 		return m.pauseOther(chid)
 	}
-	return m.resumeOther(chid)
+	err := m.resumeOther(chid)
+	if err != nil {
+		return err
+	}
+	chst, err := m.channels.GetByID(context.TODO(), chid)
```
[the remark this answers is not in the document]
Comment (hannahhoward) [reply impl/events.go:218]: This is a long standing issue but we really don't have a context for the the data transfer manager and we don't use the context passed to OpenDataTransferChannel well. Will fix seperately.

**Comment by hannahhoward** [comment]
@rvagg can I get an OK to merge from you?

File: go.mod [hunk 3]
```diff
@@ -10,34 +10,32 @@ require (
...
 	github.com/ipfs/go-ipfs-delay v0.0.1
 	github.com/ipfs/go-ipfs-exchange-offline v0.1.1
 	github.com/ipfs/go-ipfs-files v0.0.8
-	github.com/ipfs/go-ipld-cbor v0.0.5
 	github.com/ipfs/go-ipld-format v0.2.0
 	github.com/ipfs/go-log/v2 v2.5.1
 	github.com/ipfs/go-merkledag v0.5.1
 	github.com/ipfs/go-unixfs v0.3.1
-	github.com/ipld/go-ipld-prime v0.16.0
+	github.com/ipld/go-ipld-prime v0.17.1-0.20220624062450-534ccf82237d
```
Comment (rvagg) [inline go.mod:27]: we have an 0.18 we should be using now

**Review by rvagg — APPROVED** [review]
yeah, ok, a big dose of 🤞 from me but it's time to get this all merged and exercised.
the fsm stuff around the new paused states are a bit fuzzy for me, there's quite a bit of extra complexity being introduced when those states flow through the system

**Pull request merged**